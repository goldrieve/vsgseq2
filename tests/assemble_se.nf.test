nextflow_process {

    name "Test Process ASSEMBLE (single-end)"
    script "modules/assemble.nf"
    process "ASSEMBLE"

    test("assemble_se: Should assemble SE reads without failures") {

        when {
            params {
                cores = 1
                trinitymem = 10
            }
            process {
                """
                // tuple val(meta), path(trimmed)
                input[0] = [ [ id: 'late_se', single_end: true ], file("${projectDir}/data/reads/latesingle.fq.gz") ]
                input[1] = params.cores
                input[2] = params.trinitymem
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
