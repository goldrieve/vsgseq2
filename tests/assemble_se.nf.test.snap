{
    "SE: Should run without failures": {
        "content": [
            {
                "0": [
                    "latesingle.fq_trinity.Trinity.fasta:md5,f12a8901521d8026d14770a241ed5944"
                ],
                "trinity_fasta": [
                    "latesingle.fq_trinity.Trinity.fasta:md5,f12a8901521d8026d14770a241ed5944"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-20T13:12:54.185721972"
    }
}